gremlin/base_classes.py

```
@@ -84,11 +84,11 @@ class AbstractActionData(ABC):
     # Fields expected to be present in every action plugin. These define
     # information required by the plugin manager as well as UI and behavior
     # logic.
-    version : int = -1
-    name : str = ""
-    tag : str = ""
-    icon : str = ""
-    functor = None
+    version : int
+    name : str
+    tag : str
+    icon : str
+    functor: Type[AbstractFunctor]
```

May need to think of whether having defaults is desirable.

---

Defaults that are placeholder values is useful if we check for those - if we're not checking, then better to not have defaults, that might give a more obscure error, possibly later in code execution. Just my 2c.

---

Hmmm, yes could argue that if they're not there one might not use them for verification either :-) Though the reason I added them I think was to be able to quickly test if them being defined that way gets modified as intended. So not having defaults is fine here since it's not been used in the past anyway.